python-wrapper/src/neo4j_viz/gql_create.py

```diff
@@ -0,0 +1,337 @@
+                    in_string = ch
+                elif ch in ["{", "["]:
+                    depth += 1
+                elif ch in ["}", "]"]:
+                    depth -= 1
+                elif ch == "," and depth == 0:
+                    pair = prop_str[start_idx:i].strip()
+                    if ":" not in pair:
+                        snippet = _get_snippet(query, prop_start + start_idx)
+                        raise ValueError(f"Property syntax error near: `{snippet}`.")
+                    k, v = pair.split(":", 1)
+                    k = k.strip().strip("'\"")
+
+                    if k in top_level_keys:
+                        top_level[k] = _parse_value(v)
+                    else:
+                        props[k] = _parse_value(v)
+
+                    start_idx = i + 1
+            else:
+                if ch == in_string:
+                    in_string = None
+
+        if prop_str[start_idx:]:
+            pair = prop_str[start_idx:].strip()
+            if ":" not in pair:
+                snippet = _get_snippet(query, prop_start + start_idx)
+                raise ValueError(f"Property syntax error near: `{snippet}`.")
+            k, v = pair.split(":", 1)
+            k = k.strip().strip("'\"")
+
+            if k in top_level_keys:
+                top_level[k] = _parse_value(v)
+            else:
+                props[k] = _parse_value(v)
+
+        return top_level, props
+
+    def parse_labels_and_props(
+        s: str, top_level_keys: set[str]
+    ) -> tuple[Optional[str], dict[str, Any], dict[str, Any]]:
+        prop_match = re.search(r"\{(.*)\}", s)
+        prop_str = ""
+        if prop_match:
+            prop_str = prop_match.group(1)
+            prop_start = query.index(prop_str, query.index(s))
+            s = s[: prop_match.start()].strip()
+        alias_labels = re.split(r"[:&]", s)
+        raw_alias = alias_labels[0].strip()
+        final_alias = raw_alias if raw_alias else None
+
+        if prop_str:
+            top_level, props = parse_prop_str(prop_str, prop_start, top_level_keys)
+        else:
+            top_level = {}
+            props = {}
+
+        label_list = [lbl.strip() for lbl in alias_labels[1:]]
+        if "labels" in props:
+            props["__labels"] = props["labels"]
+        props["labels"] = sorted(label_list)
+
+        return final_alias, top_level, props
+
+    nodes = []
+    relationships = []
+    alias_to_id = {}
+    anonymous_count = 0
+
+    query = re.sub(r"(?i)^create\s*", "", query, count=1).rstrip(";").strip()
+    parts = []
+    paren_level = 0
+    bracket_level = 0
+    current: list[str] = []
+    for i, char in enumerate(query):
+        if char == "(":
+            paren_level += 1
+        elif char == ")":
+            paren_level -= 1
+            if paren_level < 0:
+                snippet = _get_snippet(query, i)
+                raise ValueError(f"Unbalanced parentheses near: `{snippet}`.")
```

Comment:
thats the logic we could reuse above as well

Reply:
We could but we don't need to since we already do it here